### PR TITLE
feat: create input component

### PR DESCRIPTION
**Pull request by paulgrym**
## Jira issue

JIRA issue: [FSU-78](https://klau.atlassian.net/browse/FSU-78)

## Describe your changes
Input component added

## Checklist before requesting a review

- [x] Checked which branch you would merge.
- [x] Checked if the code follows the style guidelines of this project (use eslint and prettier).
- [x] Performed a self-review of code (bugs, clean code rules, etc).
- [x] Checked if added variables have unique names.
- [x] Checked if added code/function is not a duplicate. 
- [x] Tests have been added if possible.
- [x] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: packages/design-system/theme/theme.js [hunk 1]
```diff
@@ -42,6 +42,8 @@ const theme = {
     },
     decorative: {
       100: "#0D99FF",
+      success: "#14AE5C",
```
Comment (im-klau) [inline packages/design-system/theme/theme.js:45]: Just in order to keep consistency, I'd follow the number rule, so:
```jsx
    decorative: {
      100: "#0D99FF",
      success: {
         100:  "#14AE5C"
     }
```

I'm not sure what to do with the `decorative[100]` color though 🤔

File: packages/design-system/components/icon/Icon.jsx [hunk 2]
```diff
@@ -57,6 +59,8 @@ export const icons = {
   heartOutline: HeartOutline,
   cat: Cat,
   dog: Dog,
+  error: Error,
+  success: Success,
```
Comment (im-klau) [inline packages/design-system/components/icon/Icon.jsx:63]: I think there's a `done` icon already, which probably can be used in the input valid state 🤔

Comment (paulgrym) [reply packages/design-system/components/icon/Icon.jsx:63]: done icon has less free space around the actual icon 😓

File: packages/design-system/theme/theme.js (hunk 1 above)
Comment (paulgrym) [reply packages/design-system/theme/theme.js:45]: ok, changed 😄

**Review by im-klau — CHANGES_REQUESTED** [review]

File: packages/design-system/components/input/__tests__/Input.test.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+const label = "Label"
+const requiredMessage = "Pole wymagane"
+const message = "Test message"
+
+describe(`Input`, () => {
+  it(`renders with default properties`, () => {
+    render(<Input label={label} />)
+    expect(screen.getByText(label)).toBeInTheDocument()
+    expect(screen.getByRole("textbox")).toBeInTheDocument()
```
Comment (im-klau) [inline packages/design-system/components/input/__tests__/Input.test.jsx:15]: wooaa `textbox`, nice!! 👏

File: packages/design-system/components/input/__tests__/Input.test.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        placeholder="Example placeholder text"
+        state="valid"
+      />
+    )
+    expect(screen.getByRole("textbox")).toBeInTheDocument()
+  })
+
+  it(`renders disabled when disabled attribute passed`, () => {
+    render(
+      <Input label={label} disabled placeholder="Example placeholder text" />
```
Comment (im-klau) [inline packages/design-system/components/input/__tests__/Input.test.jsx:32]: I feel like there's not much sense in passing props that aren't needed for a particular test case. I mean the placeholder 🤪 

But overall, I love all of your tests! 😍

File: packages/design-system/theme/theme.js [hunk 5]
```diff
@@ -42,6 +42,12 @@ const theme = {
     },
     decorative: {
```
Comment (im-klau) [inline packages/design-system/theme/theme.js:43]: Oh, one more thing about the theme colors - they don't work in Storybook anymore 😂 
https://design-system-git-fsu-78-input-fsu.vercel.app/?path=/story/theme--page

File: packages/design-system/components/input/Input.jsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  state,
+  message,
+  required,
+  disabled,
+  type,
+  name,
+  id,
+}) => (
+  <Styled.Label>
+    <Styled.LabelText>{label}</Styled.LabelText>
```
Comment (im-klau) [inline packages/design-system/components/input/Input.jsx:18]: shouldn't this be the Typography component with a proper variant? 🤔

File: packages/design-system/components/input/Input.jsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        name={name}
+        id={id}
+      />
+      {state && (
+        <Styled.Icon state={state}>
+          <Icon name={state === "valid" ? "success" : "error"} size="medium" />
+        </Styled.Icon>
+      )}
+    </Styled.Container>
+    <Styled.Message required={required} data-testid="message">
```
Comment (im-klau) [inline packages/design-system/components/input/Input.jsx:35]: shouldn't this be the Typography component with a proper variant? 🤔

File: packages/design-system/components/input/Input.styled.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        return theme.colors.decorative.error[100]
+      default:
+        return theme.colors.grey[400]
+    }
+  }};
+`
+
+export const Message = styled.p`
+  margin: 0;
+  min-height: 22.4px;
```
[the remark this answers is not in the document]
Comment (paulgrym) [reply packages/design-system/components/input/Input.styled.js:75]: Ok, good idea :D

File: packages/design-system/components/input/__tests__/Input.test.jsx (hunk 4 above)
Comment (paulgrym) [reply packages/design-system/components/input/__tests__/Input.test.jsx:32]: You are right, placeholder prop wasn't supposed to be here 😅

File: packages/design-system/components/input/Input.jsx (hunk 6 above)
Comment (paulgrym) [reply packages/design-system/components/input/Input.jsx:18]: Yes, it should 😅 I totally forgot about that component 🤦‍♀️

File: packages/design-system/theme/theme.js (hunk 5 above)
Comment (paulgrym) [reply packages/design-system/theme/theme.js:43]: I see what you mean - I will look closely at our theme colors gallery code 🕵️‍♀️

File: packages/design-system/components/icon/Icon.jsx (hunk 2 above)
Comment (im-klau) [reply packages/design-system/components/icon/Icon.jsx:63]: Oh, probably they should be unified somehow, but let's leave the new one as it is. We'll get back to this issue...next year 😂 And we'll solve it in a separate PR

File: packages/design-system/components/input/Input.jsx (hunk 7 above)
Comment (paulgrym) [reply packages/design-system/components/input/Input.jsx]: Now I have one problem - how to pass data-testId to message when I use Typography component? 😅

**Review by im-klau — APPROVED** [review]

File: packages/design-system/components/input/__stories__/Input.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+<Canvas>
+  <Story
+    name="Overview"
+    args={{
+      label: "Example label",
+      type: "text",
+      placeholder: "test",
+    }}
+  >
+    {Template.bind()}
```
Comment (nata-kab) [inline packages/design-system/components/input/__stories__/Input.stories.mdx:18]: Looks good 😀 You could use the Canvas for each component, this is the best practice to show multiple examples in Storybook .

**Comment by nata-kab** [comment]
Well done! 👏 🚀

File: packages/design-system/components/input/__stories__/Input.stories.mdx (hunk 9 above)
Comment (im-klau) [reply packages/design-system/components/input/__stories__/Input.stories.mdx:18]: oooh, nice catch Natalia! 

I totally agree, one Canvas per component should be better 🙈 
Zrzut ekranu 2023-01-4 o 14 16 53

**Review by AmandaKr — APPROVED** [review]
You have done great job, code is clean and everything works fine 🚀 !

**Pull request merged**